node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            .set_arbitrary_id_stamp(agent_id_stamp)
+            .service_fee_balance_minor_result(service_fee_balance_in_minor);
+        let adjustment_setup = PreparedAdjustment {
+            agent: Box::new(agent),
+            adjustment_analysis: AdjustmentAnalysis::new(
+                Adjustment::TransactionFeeInPriority {
+                    affordable_transaction_count: 2,
+                },
+                analyzed_payables,
+            ),
+            response_skeleton_opt: None,
+        };
+
+        let result = subject.adjust_payments(adjustment_setup, now).unwrap();
+
+        let expected_accounts = {
+            let mut account = account_1;
+            account.bare_account.balance_wei = service_fee_balance_in_minor;
+            vec![account.bare_account]
+        };
+        assert_eq!(result.affordable_accounts, expected_accounts);
+        assert_eq!(result.response_skeleton_opt, None);
+        assert_eq!(result.agent.arbitrary_id_stamp(), agent_id_stamp);
+        let log_msg = format!(
+            "DEBUG: {test_name}: \n\
+|Payable Account                            Balance Wei
+|
+|                                           Original
+|                                           Adjusted
+|
+|0x0000000000000000000000000000000000616263 100,000,000,000,000,000,000
+|                                           90,000,000,000,000,000,000
+|
+|Ruled Out Accounts                         Original
+|
+|0x0000000000000000000000000000000000646566 500,000,000,000,000,000,000
+|0x0000000000000000000000000000000000676869 250,000,000,000,000,000,000"
+        );
+        TestLogHandler::new().exists_log_containing(&log_msg.replace("|", ""));
+        test_inner_was_reset_to_null(subject)
+    }
+
+    #[test]
+    fn late_error_after_transaction_fee_adjustment_but_rechecked_transaction_fee_found_fatally_insufficient(
+    ) {
+        init_test_logging();
+        let test_name = "late_error_after_transaction_fee_adjustment_but_rechecked_transaction_fee_found_fatally_insufficient";
+        let now = SystemTime::now();
+        let balance_1 = multiple_by_billion(500_000_000_000);
+        let account_1 =
+            make_plucked_qualified_account("abc", balance_1, 300_000_000_000, 100_000_000_000);
+        // This account is eliminated in the transaction fee cut
+        let balance_2 = multiple_by_billion(111_000_000_000);
+        let account_2 =
+            make_plucked_qualified_account("def", balance_2, 50_000_000_000, 10_000_000_000);
+        let balance_3 = multiple_by_billion(300_000_000_000);
+        let account_3 =
+            make_plucked_qualified_account("ghi", balance_3, 150_000_000_000, 50_000_000_000);
+        let mut subject = PaymentAdjusterReal::new();
+        subject.logger = Logger::new(test_name);
+        let disqualification_arbiter = DisqualificationArbiter::default();
+        let disqualification_limit_2 =
+            disqualification_arbiter.calculate_disqualification_edge(&account_2);
```

Comment:
You don't need to calculate it by using the arbiter, instead you can take it out of the account.